Python3 and other fixes

JuiP opened the pull request:
Tested
Errors- none
- [x] Fix noisy logs
- [x] Fix Gtk-WARNINGS
- [x] Fix Errors in _object_chooser when no object is selected
- [x] Port to python 3
- [x] Added new translations to hi.po
- [x] Refreshed the pot file, fetched new translations and deleted the empty ones

@chimosky @quozl Please test, review and merge :)

JuiP committed:
Fix noisy logs

JuiP committed:
Fix Gtk-WARNING: Can't set a target list till gtk_drag_set() is called

JuiP committed:
Run 2to3

JuiP committed:
update exec and replace statvfs with disk_usage as statvfs is removed after python 3.3

JuiP committed:
sugarpycha: change callable to isinstance and more changes

JuiP committed:
flake8 fix

JuiP committed:
Fix traceback when no object is selected in _object_chooser

JuiP committed:
Fetch translations and delete empty translations

JuiP committed:
refresh pot file

JuiP committed:
Add .flake8, not added in flake8 fix

JuiP committed:
Added Screenshots

JuiP committed:
Rename screenshots

JuiP committed:
Update README.md

chimosky reviewed (changes requested):
You can demote logging instead of removing them entirely, you can demote to `debug`.

chimosky commented on readers.py:
```diff
@@ -21,7 +21,7 @@
 import os
 import glob
-import statvfs
+import shutil
```
For consistency you should use `os.statvfs`, they're various uses of that in sugar code base and none of `shutil.disk_usage`.

JuiP replied:
Done in `1f1bb36`

chimosky commented:
Thanks, reviewed 00e5df3.

JuiP committed:
demote logging

JuiP committed:
Update exec and use os.statvfs instead of shutil.disk_usage

JuiP commented:
@chimosky changed as suggested!

quozl commented:
@Hrishi1999, is further work on AnalyzeJournal compatible with your [plan for dashboard](https://github.com/sugarlabs/sugar/pull/851)?  Or is there something you can use here?

chimosky commented:
Tested, reviewed. Thanks.

In 1f1bb36, I think using `stat.f_bsize` is better than indexing to access them as it makes it easier at a glance to know which member of the `statvfs` structure is used.

Is there a particular reason why you used `collections.Callable` in 97fdd0f and not the builtin `callable`? 

The object chooser created by the `Read Turtle data` button doesn't have focus until activity closes, also occurs at master. Can you take a look?, thanks.

Logs show this error.
```
1586012753.417484 WARNING root: No icon with the name line was found in the theme.
1586012753.426675 WARNING root: No icon with the name line was found in the theme.
```

Hrishi1999 commented:
@quozl The Dashboard Activity was supposed to be an extension of AnalyzeJournal integrated with Sugar's Journal. Dashboard does use sugarpycha, but the version of sugarpycha in Dashboard is already in python3. In future, to avoid conflicts, Dashboard can use AnalyzeJournal's sugarpycha version as a submodule or vice-versa.

JuiP commented:
Thanks for testing :)
> Is there a particular reason why you used collections.Callable in 97fdd0f and not the builtin callable?

I saw chart activity used sugarpycha which was using collections.Callable instead of builtin callable when it was ported to python3 , so I just used it to maintain similarity in the code of sugarpycha

> In 1f1bb36, I think using stat.f_bsize is better than indexing to access them as it makes it easier at a glance to know which member of the statvfs structure is used.

Yes, I agree with you. Except for making the code more readable it isn't much of a difference. I will make the change if changes to AnalyzeJournal are compatible. Should I go ahead with changes @quozl @chimosky ? 

> The object chooser created by the Read Turtle data button doesn't have focus until activity closes, also occurs at master. Can you take a look?, thanks.

Doesn't `c02848e` fix this problem? Could you please explain what you meant by doesn't have focus?

> Logs show this error.

Sorry, I didn't consider that as an error. It was a warning I didn't know the cause of. Any suggestions what might have caused it / what could probably fix it ?

Hrishi1999 commented:
Tested, https://github.com/sugarlabs/AnalyzeJournal/pull/11/commits/1f1bb362fd70c394d4dcf596516ee0d76a12f0d7 
The problem with ObjectChooser does exist. It doesn't have focus, that is, ObjectChooser appears behind the activity, you will see the ObjectChooser only when you close the activity or tab our.

quozl commented:
> @quozl The Dashboard Activity was supposed to be an extension of AnalyzeJournal integrated with Sugar's Journal. Dashboard does use sugarpycha, but the version of sugarpycha in Dashboard is already in python3. In future, to avoid conflicts, Dashboard can use AnalyzeJournal's sugarpycha version as a submodule or vice-versa.

Sorry, my question must have been a bit too vague.  What I'm saying is that we seem to have two parallel developments to choose from; either a Dashboard in Sugar, or the AnalyzeJournal activity being ported to Python 3 so it can be used in Sugar again.  I'm fine with competition to choose the best solution, but would it be more effective to bring the AnalyzeJournal features into the draft Dashboard feature, or to take some of the draft Dashboard features and bring them into AnalyzeJournal?

Hrishi1999 commented:
Sorry, it seems my last answer wasn't focused on the question. I would like the features to be ported to Dashboard instead since it is going to be integrated with the Journal. But there are some features in AnalyzeJournal which may not make sense if we implement in the Dashboard. 
@chimosky Do you have any suggestions?

chimosky commented:
I agree with @Hrishi1999, some features of `AnalyzeJournal` should be ported to Dashboard not all features.

chimosky commented:
> Doesn't c02848e fix this problem? Could you please explain what you meant by doesn't have focus?

Tested again and the issue no longer persists, when I said "doesn't have focus" I meant when the button is clicked, the object chooser window didn't come up like it's supposed to but came up behind the activity canvas and was only visible when the activity was closed.

> Yes, I agree with you. Except for making the code more readable it isn't much of a difference. I will make the change if changes to AnalyzeJournal are compatible. Should I go ahead with changes @quozl @chimosky ?

Yes it doesn't make any functional difference, my earlier comment was for readability. 
Yes you can go on with the change, thanks.

> Sorry, I didn't consider that as an error. It was a warning I didn't know the cause of. Any suggestions what might have caused it / what could probably fix it ?

The warning is caused when `_get_icon_info` of `sugar3.graphics.icon._IconBuffer` is called with just an `icon_name` arg and a `Gtk.IconTheme.lookup` is performed and the query returns `None`.
The `lookup` is performed on an icon with the name `line` and there's no icon with such name in `icons/`, you'll have to figure out why it happens.
Hope that helps.

srevinsaju commented on README.md:
```diff
@@ -1,15 +1,23 @@
+===============
+This Activity charts data from the Sugar Journal and gives you a visual representation of Disk usage, Activity usage and Turtle data usage using a vertical bar chart, a horizontal bar chart or a pie chart.
 
-This Activity charts data from the Sugar Journal:
-
-1. free space
-2. activity usage
-3. turtle block usage
-
-This activity gives you the possibility to graphically, the journal usage.
-
-### To Know more about AnalyzeJournal and Sugar, Please refer to ###
```
Is it necessary to remove this?

JuiP replied:
It wasn't exactly removed just rephrased.

srevinsaju replied:
Ig most of the information is not clearly mentioned in the new Readme. You can mark it as resolved if you feel its right.

JuiP replied:
If you have any suggestions or a better description, I would love to add it if its well-fitting! Otherwise I feel the readme is alright! :)

srevinsaju commented on activity.py:
```diff
@@ -412,39 +412,39 @@ def _graph_from_reader(self, reader):
-            reader = TurtleReader(file_path)
-            self._graph_from_reader(reader)
+        try:
+            matches_mime_type, file_path, title = self._object_chooser('application/x-turtle-art', _('Turtle'))
```
Maybe you can split the line into two, exceeds character limit (79), on the line. (E501, Flake) . Atm it has 111

JuiP replied:
Done in 8181469

srevinsaju commented on activity.py:
```diff
@@ -519,15 +518,12 @@ def read_file(self, file_path):
-    __gsignals__ = {
-             'label-changed': (GObject.SignalFlags.RUN_FIRST, None, [str, str], ),
-             'value-changed': (GObject.SignalFlags.RUN_FIRST, None, [str, str], ), }
+    __gsignals__ = {'label-changed': (GObject.SignalFlags.RUN_FIRST, None, [str, str], ), 'value-changed': (GObject.SignalFlags.RUN_FIRST, None, [str, str], ), }
```
Split it into multiple lines. Seems like this was extra diff. The old code was equally good.

JuiP replied:
Done in 8181469

srevinsaju commented on readers.py:
```diff
@@ -95,53 +94,11 @@ class TurtleReader():
-            'extrasp']
-    PALNAMES = [_('turtle'), _('pen'), _('number'), _('flow'), _('box'),
-                _('sensor'), _('media'), _('extras')]
+    TACAT = {'clean': 'forward', 'forward': 'forward', 'back': 'forward', 'left': 'forward', 'right': 'forward', 'arc': 'arc', 'xcor': 'coord', 'ycor': 'coord', 'heading': 'coord', 'setxy2': 'setxy', 'seth': 'setxy', 'penup': 'pen', 'pendown': 'pen', 'setpensize': 'pen', 'setcolor': 'pen', 'pensize': 'pen', 'color': 'pen', 'setshade': 'pen', 'setgray': 'pen', 'shade': 'pen', 'gray': 'pen', 'fills ...
+    TAPAL = {'forward': 'turtlep', 'arc': 'turtlep', 'coord': 'turtlep', 'setxy': 'turtlep', 'pen': 'penp', 'fill': 'penp', 'number': 'numberp', 'random': 'numberp', 'boolean': 'numberp', 'repeat': 'flowp', 'ifthen': 'flowp', 'action': 'boxp', 'box': 'boxp', 'sensor': 'sensorp', 'media': 'mediap', 'extras': 'extrasp', 'python': 'extrasp'}
+    TASCORE = {'forward': 3, 'arc': 3, 'setxy': 2.5, 'coord': 4, 'turtlep': 5, 'pen': 2.5, 'fill': 2.5, 'penp': 5, 'number': 2.5, 'boolean': 2.5, 'random': 2.5, 'numberp': 0, 'repeat': 2.5, 'ifthen': 7.5, 'flowp': 10, 'box': 7.5, 'action': 7.5, 'boxp': 0, 'media': 5, 'mediap': 0, 'python': 5, 'extras': 5, 'extrasp': 0, 'sensor': 5, 'sensorp': 0}
+    PALS = ['turtlep', 'penp', 'numberp', 'flowp', 'boxp', 'sensorp', 'mediap', 'extrasp']
+    PALNAMES = [_('turtle'), _('pen'), _('number'), _('flow'), _('box'), _('sensor'), _('media'), _('extras')]
```
Lines extremely long, try prettifying dictionaries for better code readability

JuiP replied:
Done in 8181469

srevinsaju reviewed (changes requested):
The tables might not be evenly spaced. On launching the activity. it looks like this
![image](https://user-images.githubusercontent.com/48695438/78636984-22b4af80-78b2-11ea-926d-dc1410b583fe.png)


These are the tracebacks I got
![image](https://user-images.githubusercontent.com/48695438/78637400-f3eb0900-78b2-11ea-8aa4-d48cde17eabf.png)
![image](https://user-images.githubusercontent.com/48695438/78637443-06fdd900-78b3-11ea-8f1d-0a1454b07947.png)

~I guess @quozl mentioned some of the errors previously~ Nvm

Saumya-Mishra9129 reviewed (commented):
> 1586012753.417484 WARNING root: No icon with the name line was found in the theme.
1586012753.426675 WARNING root: No icon with the name line was found in the theme.

@JuiP I had a quick review of code and found that the code first when written thought to implement line chart of Sugarpycha also along with pie, vbar and hbar. As look at charts.py Line [121](https://github.com/sugarlabs/AnalyzeJournal/blob/6ae7ae6e96d137449135439a4d69375ebbe628aa/charts.py#L121)  and activity.py line [483.](https://github.com/sugarlabs/AnalyzeJournal/blob/6ae7ae6e96d137449135439a4d69375ebbe628aa/activity.py#L483). But not adding line chart in [activity.py ](https://github.com/sugarlabs/AnalyzeJournal/blob/6ae7ae6e96d137449135439a4d69375ebbe628aa/activity.py#L171) causing the [error ](https://github.com/sugarlabs/AnalyzeJournal/pull/11#pullrequestreview-388809906)as seen by @srevinsaju  and above mentioned error also. Thanks :)

JuiP commented:
Thanks @Saumya-Mishra9129 will get back to working soon, for now I have been working on too many college assignments and projects due by this friday.

JuiP committed:
Added line chart

- Fixes the WARNING root: No icon with the name line was found in the theme
- Adds a feature: Line chart to the activity

Saumya-Mishra9129 reviewed (commented):
Reviewed 4d4eb12, looks fine to me :)

JuiP commented:
@srevinsaju how do I reproduce the error you mentioned? Check if 4d4eb12 fixes it for you?

> Tested again and the issue no longer persists, when I said "doesn't have focus" I meant when the button is clicked, the object chooser window doesn't come up like it's supposed to but came up and went behind the activity canvas and was only visible when the activity was closed.

On testing again, this issue is still there for me. If someone has tested activities which use ObjectChooser( I think deducto uses it), can you confirm if the same issue is seen in those activities too? If it is then the fix should be made in  [sugar-toolkit-gtk3](https://github.com/sugarlabs/sugar-toolkit-gtk3/blob/master/src/sugar3/graphics/objectchooser.py) @chimosky @quozl

Saumya-Mishra9129 commented:
Reviewed again 4d4eb12, when we click on Read Turtle Data button, Object chooser window doesnt appear in front , it is behind of canvas window thats why we are not able to see that time and when we click on stop button we can see object chooser window , it means it was hidden.
Also I am getting this error,
![Screenshot from 2020-05-12 16-34-09](https://user-images.githubusercontent.com/45252461/81677762-d7705c80-946e-11ea-8122-cf1eb2bb9c4a.png)

JuiP commented:
Hey @Saumya-Mishra9129, can you please tell me how did you get that error? I have nothing in my logs, also do this error also occur when you test master?

srevinsaju commented:
I tested  https://github.com/sugarlabs/AnalyzeJournal/pull/11/commits/4d4eb129759671ec321470acfda7306e6578058a. ~Seems like the ObjectChooser is fixed now (only for LoadTurtle file)~.  See https://github.com/sugarlabs/AnalyzeJournal/pull/11#issuecomment-627276382

However, this remains unfixed
![image](https://user-images.githubusercontent.com/48695438/81678497-4becd080-945a-11ea-9d43-5ba88092acc8.png)

srevinsaju commented:
Interesting results. Seems like the bug for Objectchooser is unfixed, hinting it might be a bug in the call of ObjectChooser in the code. (Please verify the signature of the functions have been / not been changed over)

Simple reproducer:
* Restart Sugar
* Clone and install ColorDeducto
* Run colordeducto, test the Object Chooser
* Close colordeduct
* Open AnalyzeJournal
* Test ObjectChooser, It works fine
------------
* Restart sugar
* open analyzeJournal
* Try clicking the Turtle Load option
* You will notice no window is seen

JuiP commented:
> Seems like the ObjectChooser is fixed now

Well that's weird, it isn't fixed for me and Saumya. Before 4d4eb12, it was fixed for me.

@srevinsaju please tell me your test method for getting the error.

srevinsaju commented:
@JuiP 
> Well that's weird, it isn't fixed for me and Saumya. Before 4d4eb12, it was fixed for me.

See https://github.com/sugarlabs/AnalyzeJournal/pull/11#issuecomment-627276382

srevinsaju commented:
To get the `ValueError` on https://github.com/sugarlabs/AnalyzeJournal/pull/11#issuecomment-627274781, click this

![image](https://user-images.githubusercontent.com/48695438/81680488-cb2ed400-945b-11ea-927b-8295082599df.png)


-------------------------

To get the IndexError, you have to click `Save Image` option 
![image](https://user-images.githubusercontent.com/48695438/81680156-77bc8600-945b-11ea-84ed-15da91d0de90.png)
![image](https://user-images.githubusercontent.com/48695438/81680277-9458be00-945b-11ea-8965-ee87cdacd1f2.png)



-------------

> Hey <> , can you please tell me how did you get that error? I have nothing in my logs, also do this error also occur when you test master?

@JuiP, is your logging system alright? 
* Please configure sugar in DEBUG mode to make sure you get all the logs if you are not running within terminal. 
* Make sure all the methods / functions are completely tested before next review
* To test the code, please do a coverage test. If you are getting a coverage percentage below 85%, you might need to expect more bugs in your code.

See the coverage guide in sugar-docs

JuiP commented:
![coverage](https://user-images.githubusercontent.com/46154968/81688975-8dd83f80-9477-11ea-8533-bc9e55faac39.png)

I think the coverage is low because I can't test the Turtle file as the object chooser appears behind the window. I have tried clicking all sorts of buttons, and saving each graph as image.

> Please configure sugar in DEBUG mode to make sure you get all the logs if you are not running within terminal.

I am running from within the sugar-desktop terminal with command sugar-activity3, I tried doing what you displayed in the images, tried read Freespace Data and save as image. I did not get the tracebacks. The image is saved as png in my journal. I don't understand why is there a difference when testing. I am using a debian build (If that helps)

srevinsaju commented:
@JuiP I am unable to save a png of the graph for some reason. 
Did you try [this](https://github.com/sugarlabs/sugar/blob/master/docs/development-environment.md#change-debugging-level) to enable logging?

I might have been suspicious if the error I had received might be because of something wrong with my system; but as @Saumya-Mishra9129 were also able to reproduce it (1) [here](https://github.com/sugarlabs/AnalyzeJournal/pull/11#issuecomment-627271565), so I was quite suspicious if something had gone wrong during the Port

Regarding https://github.com/sugarlabs/AnalyzeJournal/pull/11#issuecomment-627274781, I can confirm that this is a bug which possibly happened due to python3 port (I have removed all of python2 from my system, so can't test `master`).

An immediate correction on 

https://github.com/JuiP/AnalyzeJournal/blob/python3/activity.py#L573-L576

would be to add another Exeception catch 

change this 
```python
        try:
            _iter = self.model.insert(path, [label, value])
        except ValueError:
            _iter = self.model.append([label, str(value)])
```

to 
```python
        try:
            _iter = self.model.insert(path, [label, value])
        except (ValueError, TypeError):
            _iter = self.model.append([label, str(value)])
```

In the long run, you might need to find the fix for why it actually comes out as `int` or `float` instead of `str`; But the above patch would fix the error, in a legal way.

I am not sure, however of [this error](https://github.com/sugarlabs/AnalyzeJournal/pull/11#issuecomment-627271565) and the ObjectChooser Error by simply looking at code, (might need to debug and compare other upstream activities to see what makes them work, and not this)

Hrishi1999 commented:
Tested https://github.com/sugarlabs/AnalyzeJournal/pull/11/commits/4d4eb129759671ec321470acfda7306e6578058a. I couldn't seem to reproduce the tracebacks provided by @Saumya-Mishra9129 and @srevinsaju. One weird thing to notice is that ObjectChooser only worked once for me, after rebooting the system, it was only visible when the activity is closed.

Saumya-Mishra9129 commented:
> I tested [4d4eb12](https://github.com/sugarlabs/AnalyzeJournal/commit/4d4eb129759671ec321470acfda7306e6578058a). ~Seems like the ObjectChooser is fixed now (only for LoadTurtle file)~. See [#11 (comment)](https://github.com/sugarlabs/AnalyzeJournal/pull/11#issuecomment-627276382)
> 
> However, this remains unfixed
> ![image](https://user-images.githubusercontent.com/48695438/81678497-4becd080-945a-11ea-9d43-5ba88092acc8.png)

I am not getting this error as mentioned by @srevinsaju , but I am getting index error.

Saumya-Mishra9129 commented:
@JuiP Add line chart [here](https://github.com/sugarlabs/AnalyzeJournal/blob/6ae7ae6e96d137449135439a4d69375ebbe628aa/activity.py#L193) as well . I think It can help for [index](https://github.com/sugarlabs/AnalyzeJournal/pull/11#issuecomment-627271565) error.

srevinsaju commented:
@Saumya-Mishra9129 I were talking about IndexError

I have two errors:
* TypeError (only by me)
* IndexError  (faced by you and me)

JuiP commented:
> @JuiP Add line chart here as well . I think It can help for index error.
Well yes this seems to be the most likely cause for the error, can you test the change:

```
self.chart_type_buttons = [add_vbar_chart,
                                   add_hbar_chart,
                                   add_line_chart,
                                   add_pie_chart]
```
@Saumya-Mishra9129 You will have to test this patch for me before I add it to the commit because I am unable to reproduce the traceback. I'm not sure why, but logically if this fixes it then I should have got the traceback too :confused: 

> An immediate correction on would be to add another Exeception catch

Yes, that could be the most intuitive way to fix it but I'll wait till I find out the cause, because none of us seem to get this error in the logs. @srevinsaju

Saumya-Mishra9129 commented:
> self.chart_type_buttons = [add_vbar_chart,
                                   add_hbar_chart,
                                   add_line_chart,
                                   add_pie_chart]

@JuiP I tested it, working fine not getting index error anymore.

srevinsaju commented:
```bash
rm -r ~/.sugar
```
fixed my problem :rofl:

chimosky commented:
> On testing again, this issue is still there for me. If someone has tested activities which use ObjectChooser( I think deducto uses it), can you confirm if the same issue is seen in those activities too? If it is then the fix should be made in [sugar-toolkit-gtk3](https://github.com/sugarlabs/sugar-toolkit-gtk3/blob/master/src/sugar3/graphics/objectchooser.py) @chimosky @quozl


The problem isn't from the toolkit, you should specify the `parent` kwarg as `self` when `ObjectChooser` is called, also add an else statement to the if statement in `self._object_chooser`that binds `file_path` to `None`.

> Please configure sugar in DEBUG mode to make sure you get all the logs if you are not running within terminal.

@srevinsaju You don't need to have sugar in DEBUG mode to see logs.

>  am running from within the sugar-desktop terminal with command sugar-activity3, I tried doing what you displayed in the images, tried read Freespace Data and save as image. I did not get the tracebacks. The image is saved as png in my journal. I don't understand why is there a difference when testing. I am using a debian build (If that helps)

You should run the activity from the activity ring as that's how a user would run it and not from the terminal.

JuiP committed:
Added improvements:

- Fixes index error : reported by @Saumya-Mishra9129 and @srevinsaju
- Fixed : Object Chooser focus

JuiP committed:
Flake8 fix for E501: line too long

JuiP commented:
> The problem isn't from the toolkit, you should specify the parent kwarg as self when ObjectChooser is called

Thanks for the hint @chimosky ! Fixed it in ef16be6

> also add an else statement to the if statement in self._object_chooserthat binds file_path to None.

I have actually taken care of it by exception handling in c02848e. It does the same thing :)

> You should run the activity from the activity ring as that's how a user would run it and not from the terminal.

Sorry, I didn't make myself clear. I do run the activity from the ring and from the terminal (to test coverage or test just minor code changes)

I tested this activity, works perfectly for me @srevinsaju @Saumya-Mishra9129 @Hrishi1999 please review and test. Thanks :)

If it's approved by all then maybe this PR is good to merge? :) @chimosky @quozl ?

chimosky commented on activity.py:
```diff
@@ -381,7 +382,8 @@ def _set_chart_line_color(self, widget, pspec):
 
     def _object_chooser(self, mime_type, type_name):
-        chooser = ObjectChooser()
+        chooser = None
```
I'm guessing you forgot this.

JuiP replied:
Done in a7f1b91 :)

JuiP committed:
Changes to ObjectChooser

- Removed chooser = None assignment
- Removed what_filter kwarg

quozl commented on .flake8:
```diff
@@ -0,0 +1,7 @@
+# gi.require_version() is required before later imports
+
+ignore = E402
+
```
Is a blank line.

JuiP replied:
Done in 14f0084

quozl commented on README.md:
```diff
@@ -1,15 +1,23 @@
-# AnalyzeJournal Activity #
+AnalyzeJournal Activity
+===============
+This Activity charts data from the Sugar Journal and gives you a visual representation of Disk usage, Activity usage and Turtle data usage using a vertical bar chart, a horizontal bar chart or a pie chart.
```
Activity does not need to be capitalised.

JuiP replied:
Done in 14f0084

quozl commented on README.md:
```diff
@@ -1,15 +1,23 @@
+
+<img src="Screenshots/2.png" width="75%" title="AnalyzeJournal Activity">
+
+First select Disk usage, Activity usage or Turtle data (the one you wish to see the graphical represenation of) and then click on the type of chart you wish to see. You can also save the chart dispayed as an image by clicking on 'save as image'.
```
Typo representation not represnation.

JuiP replied:
Done in 14f0084

quozl commented on README.md:
```diff
@@ -1,15 +1,23 @@
+On Sugar Desktop systems;
+
+* [Use My Settings,](https://help.sugarlabs.org/my_settings.html) [Software Update](https://help.sugarlabs.org/my_settings.html#software-update) 
+* Use Browse to open [activities.sugarlabs.org](https://activities.sugarlabs.org/) Search for Paths, then download
```
Paths is not the right activity to search for.

JuiP replied:
Done in 14f0084

quozl commented on activity.py:
```diff
@@ -572,31 +583,30 @@ def add_value(self, label, value):
                         True)
 
-        _logger.info("Added: %s, Value: %s" % (label, value))
+        logging.debug("Added: %s, Value: %s" % (label, value))
```
Puzzling that `_logger` is created, then instead of calling it you call the module.

JuiP replied:
Done in 14f0084

quozl commented on charthelp.py:
```diff
@@ -12,12 +12,12 @@ def create_help(toolbar):
+                           'import-journal')
     helpitem.add_paragraph(_('The types of blocks used in Turtle Art.'),
-            'import-turtle')
-    helpitem.add_paragraph(_('The graph title is the same as the Activity title'))
```
Wording change doesn't seem right; "is the same" seems okay to me.

JuiP replied:
Done in 14f0084

quozl commented on icons/line.svg:
```diff
@@ -0,0 +1,34 @@
+     d="m 594.07188,325.15231 0,-360.47952 L 434.51609,52.953494 274.96046,-175.105 115.4045,109.35506 l 0,215.79736 z"
+     id="rect3757"
+     style="color:#000000;fill:&fill_color;;fill-opacity:1;stroke:&stroke_color;;stroke-width:33.73588181;stroke-linecap:square;stroke-linejoin:miter;stroke-miterlimit:4;stroke-opacity:1;stroke-dasharray:none;stroke-dashoffset:0;marker:none;visibility:visible;display:inline;overflow:visible;enable-background:accumulate" /></g></g>
+</g></svg>
```
Missing end of line marker, also in Chart activity.

JuiP replied:
I ran ` sed -i '$a\' line.svg `
If I get something like this in git diff, does it mean I have added eol marker? @quozl 
```
-}
\ No newline at end of file
+}
```

quozl replied:
GitHub shows in this discussion last line of the diff a red circle with a slash, and a red end of line mark.

You can verify with most text files by using an editor or program that will display the bytes.

Like `hexdump -C file`.  The line.svg file in Chart activity doesn't have the newline at end of file.  You can add one by typing `echo >> file`.

```log
00000450  2d 62 61 63 6b 67 72 6f  75 6e 64 3a 61 63 63 75  |-background:accu|
00000460  6d 75 6c 61 74 65 22 20  2f 3e 3c 2f 67 3e 3c 2f  |mulate" /></g></|
00000470  67 3e 0a 3c 2f 67 3e 3c  2f 73 76 67 3e           |g>.</g></svg>|
```

after

```log
00000450  2d 62 61 63 6b 67 72 6f  75 6e 64 3a 61 63 63 75  |-background:accu|
00000460  6d 75 6c 61 74 65 22 20  2f 3e 3c 2f 67 3e 3c 2f  |mulate" /></g></|
00000470  67 3e 0a 3c 2f 67 3e 3c  2f 73 76 67 3e 0a        |g>.</g></svg>.|
```

Missing newline at end of file makes life difficult for people who use command line shells.  When you use `cat` or most other programs to display a file, the shell prompt is issued on the same line as the last line of the file.  Then command recall and editing might not work.  It's amazing how such a little thing has been with us since the 1980s.  :grin:

quozl replied:
(and if you copied the file from Chart, it's not your fault!)

quozl commented on readers.py:
```diff
@@ -63,9 +61,10 @@ def get_chart_data(self):
+        path = env.get_profile_path()
+        stat = os.statvfs(path)
+        free_space = stat[0] * stat[4]
+        total_space = stat[0] * stat[2]
```
This method of accessing the statvfs result by magic numbers is not easy to understand; you should use names instead; e.g. stat.f_bsize ... try it interactively and you'll see.

JuiP replied:
Done in 14f0084

quozl commented on readers.py:
```diff
@@ -95,51 +94,52 @@ class TurtleReader():
+             'printheap': 'extras', 'clearheap': 'extras',
+             'isheapempty2': 'extras', 'chr': 'extras', 'int': 'extras',
+             'myfunction': 'python', 'userdefined': 'python',
+             'loadblock': 'python', 'loadpalette': 'python'}
```
Have these names been revalidated against the latest Turtle Art source?

JuiP replied:
Yes, I have revalidated it from [here](https://github.com/sugarlabs/turtleart-activity/blob/master/pysamples/ta-stats.py)

quozl reviewed (commented):
Reviewed.

JuiP committed:
More Changes

- fixed unnecessary capitalization and typos in README.md
- Use _logger instead of calling the module
- Change back the wording in charthelp.py
- Used stat.f_blocks and f_bavail ... instead of indexing method

JuiP committed:
Added end of line marker

JuiP commented:
Is this good to merge? Anymore suggested changes? @quozl @chimosky
